Log static graph in constructor if it is set

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #72456
* #72455

It is easier to log if static graph is set at construction time now that it is natively supported in DDP constructor, as opposed to waiting for the first iteration to finish. In some failure cases we're seeing the first iteration does not finish and thus we don't have this data which is vaulable to debug.

Differential Revision: [D34045204](https://our.internmc.facebook.com/intern/diff/D34045204/)

rohan-varma committed:
Log static graph in constructor if it is set

It is easier to log if static graph is set at construction time now that it is natively supported in DDP constructor, as opposed to waiting for the first iteration to finish. In some failure cases we're seeing the first iteration does not finish and thus we don't have this data which is vaulable to debug.

Differential Revision: [D34045204](https://our.internmc.facebook.com/intern/diff/D34045204/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
thanks for changing this

rohan-varma committed:
Update on "Log static graph in constructor if it is set"


It is easier to log if static graph is set at construction time now that it is natively supported in DDP constructor, as opposed to waiting for the first iteration to finish. In some failure cases we're seeing the first iteration does not finish and thus we don't have this data which is vaulable to debug.

Differential Revision: [D34045204](https://our.internmc.facebook.com/intern/diff/D34045204/)

[ghstack-poisoned]

rohan-varma committed:
Update on "Log static graph in constructor if it is set"


It is easier to log if static graph is set at construction time now that it is natively supported in DDP constructor, as opposed to waiting for the first iteration to finish. In some failure cases we're seeing the first iteration does not finish and thus we don't have this data which is vaulable to debug.

Differential Revision: [D34045204](https://our.internmc.facebook.com/intern/diff/D34045204/)

[ghstack-poisoned]
